src/org/rascalmpl/interpreter/env/ModuleEnvironment.java

```diff
@@ -195,7 +195,20 @@ public void extend(ModuleEnvironment other) {
 		  if (this.generalKeywordParameters == null) {
 			  this.generalKeywordParameters = new HashMap<>();
 		  }
-		  this.generalKeywordParameters.putAll(other.generalKeywordParameters);
+
+		  for (Entry<Type, List<KeywordFormal>> e : other.generalKeywordParameters.entrySet()) {
+			 List<KeywordFormal> here = Collections.unmodifiableList(this.generalKeywordParameters.get(e.getKey()));
```

Comment:
if `this.generalKeywordParameters.get(e.getKey())` returns `null` the `unmodifiableList` throws an exception.

I also think that we don't need to wrap it in an umodifiable list, either we always have an unmodifiable list in the map, or we're save in these 4 lines and know we're not manipulating it.

Reply:
right. the wrapper was a temporary test for me.